guides/doc-Managing_Hosts/topics/proc_registering-a-host-to-project-using-the-global-registration-template.adoc

```diff
@@ -0,0 +1,174 @@
+[id="registering-a-host-to-project-using-the-global-registration-template_{context}"]
+= Registering a Host to {ProjectName} Using the Global Registration Template
+
+Use the following procedure to register a host to {ProjectNameX}.
+
+With this registration method you can register hosts to {Project} by generating a `curl` command on {Project} and running this command on hosts.
+This method uses the global registration template that gives you complete control over the process of the host registration.
+
+ifeval::["{build}" != "satellite"]
+Note that you can extend the parameters by plug-ins. For more information, see https://github.com/theforeman/foreman/blob/develop/developer_docs/how_to_create_a_plugin.asciidoc.
+endif::[]
+
+.Prerequisites
+* The {Project} user that generates the `curl` command must have the `create_hosts` permission.
+* You must have root privileges on the host that you want to register.
+
+.Procedure
+. In {Project} web UI, navigate to *Hosts* > *Provisioning Templates* and locate and click *Linux registration default*.
+
+. Click the *Association* tab.
+
+. Select the operating systems that you want to register.
+
+. Navigate to *Hosts* > *Operating Systems* and click the operating system that you want to register.
+
+. Click *Submit*.
+
+. Click the *Templates* tab.
+
+. From the *Registration template* list, select *Linux registration default*.
+
+. Click *Submit*.
+
+. Navigate to *Hosts* > *All Hosts* > *Register Host*.
+
+. Optional: If you want to register hosts to {SmartProxy}, enter the following command on the {SmartProxy} to enable required modules:
++
+[options="nowrap", subs="+quotes,attributes"]
+----
+# foreman-installer --foreman-proxy-registration \
+--foreman-proxy-templates \
+--foreman-proxy-template-url 'http://{smartproxy-example-com}'
+----
+
+. What fields are required for Satellite and Foreman?
+
+. Optional: From the *Host Group* list, select the host group to associate the hosts with.
+
+. From the *Operating System* list, select the operating system of hosts that you want to register.
+
+. Optional: If you want to register hosts to a {SmartProxy}, from the *{SmartProxy}* list, select the {SmartProxy} to register hosts to.
+
+ifeval::["{build}" == "satellite"]
+. Optional: If you do not want to register hosts with Insights, from the *Insights* list, select *No*. What are the defaults for those two?
```

Comment:
@stejskalleos what are the default values for the Insights and Remote Execution settings in Satellite and Foreman? In my Satellite I just saw blank fields, maybe putting something there for visibility would help.

Reply:
Logic for these two parameters is little bit confusing (it will be replaced in 6.10 with better behavior), I'll try my best:

**Options in Form**
* `empty option` - This means that we take the (default) value from global parameters `host_registration_insights` and `host_registration_remote_execution`. See bellow _Getting value of global parameters_
* `Yes / No` - If the selected value in form is same as the value of global parameter, we **do not**  add `setup_rex|setup_rex=value` to generated command. Only when the value is different. See the table:

Selected option in Form | Global parameter | Argument added to command
--- | --- | --- 
nothing | `true` | nothing (take value from GP)
nothing | `false` | nothing (take value from GP)
`Yes` | `true` | nothing (they're same)
`No` | `false` | nothing (they're same)
`Yes` | `false` | `...&setup_rex\|insights=true`
`No` | `true` | `...&setup_rex\|insights=false`



**Getting value of global parameters**
By default both parameters are set to `true` on global level. However, users can override it on multiple levels:

- Global Level (`insights=true`, `rex=true`)
- Organization
- Location
- Host Group
- Operating System

The rule is that lower level overrides the upper levels, see the example:

- Global Level (`insights=true`, `rex=true`)
- Organization
- Location
- Host Group (`insights=false`, `rex=false`)

Values of params: `false`

Reply:
In the recording you said that this logic is going to be reworked by the 6.9 release. Did you describe the latest planned versions?

Reply:
I will not mark this step as optional and just say that users need to select either Yes or Not to avoid any confusion. It is much easier to make users choose one concrete variant than describe this complicated logic.